pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -106,653 +62,948 @@ func TestPipeline_Validate(t *testing.T) {
 		},
 		failureExpected: true,
 	}, {
-		name: "pipeline spec missing taskref and taskspec",
+		name: "valid pipeline with params, resources, workspaces, task results, and pipeline results",
 		p: &v1beta1.Pipeline{
 			ObjectMeta: metav1.ObjectMeta{Name: "pipeline"},
 			Spec: v1beta1.PipelineSpec{
-				Tasks: []v1beta1.PipelineTask{
-					{Name: "foo"},
-				},
+				Description: "this is a valid pipeline with all possible fields initialized",
```

Comment:
nice!!

Reply:
i wonder if it makes sense to have a similar failure case pipelinespec, e.g. one with a lot of different failures (i think they'll mostly all be surfaced? not sure if the logic just bails early in that case 🤔 )

Reply:
yup it does 👍

Reply:
interesting!! totally outside the scope of what you're doing, but i feel like it'd be nice to be able to build up as many errors as possible and give them all at once